### PR TITLE
Prepare initial release

### PR DESCRIPTION
**Pull request by doodlesbykumbi**
Update CHANGELOG. Add tag triggered steps to Jenkinsfile. Push to edge on master. Tag images with commit on build.

### CODE REVIEW

File: bin/publish [hunk 1]
```diff
@@ -1,15 +1,64 @@
...
+  local tag=${1}
+  image_with_version_tag="${IMAGE_NAME}:${tag}"
+  echo "Tagging and pushing ${image_with_version_tag}"
+
+  docker tag "${IMAGE_WITH_TAG}" "${image_with_version_tag}"
+  docker push "${image_with_version_tag}"
+}
+
+function main() {
+  if [[ $# -gt 0 ]]
```
Comment (izgeri) [inline bin/publish:22]: probably worth adding a comment here about this logic - it's intended to be used for non-tag-triggered flows, right? whereas the rest of it is for tags only?

Comment (doodlesbykumbi) [reply bin/publish:22]: Yup, that's it. II'll add a comment now

File: bin/publish [hunk 2]
```diff
@@ -1,15 +1,67 @@
...
+  # If an argument is passed to the script it will be read as a tag, and this single tag
+  # will be published then the script will exit. This functionality is useful for
+  # publishing single tags without involving the tag-triggered builds logic below.
+  if [[ $# -gt 0 ]]
+  then
+      publish_image $1
+      exit 0;
+  fi
+
+  if [[ -z "${TAG_NAME:-}" ]]; then
```
Comment (jonahx) [inline bin/publish:31]: Wouldn't:
```bash
if [[ -z "$TAG_NAME" ]]; then
```
be equivalent here?

File: bin/publish [hunk 3]
```diff
@@ -1,15 +1,67 @@
...
+    echo "Please supply environment variable TAG_NAME."
+    echo "If you see this error in Jenkins it means the publish script was run"
+    echo "for a build that wasn't triggered by a tag - please check publish stage conditions."
+    exit 1
+  fi
+
+  # Accumulate TAGS
+  #
+
+  local VERSION="${TAG_NAME#"v"}"
```
Comment (jonahx) [inline bin/publish:41]: quotes around the `v` shouldn't be needed:
```bash
local VERSION="${TAG_NAME#v}"
```

File: bin/publish [hunk 4]
```diff
@@ -1,15 +1,67 @@
...
+  local TAGS=(
+    "${VERSION}"
+  )
+
+  # Add less specific version tags, eg. given 1.2.3, add 1.2 and 1
+  for tag in $(gen_versions "$VERSION"); do
```
Comment (jonahx) [inline bin/publish:47]: Style inconsistency between this line and line 43.  I think you should either always use the bracket form or else only use the bracket form when needed.  In the latter case, you'd change line 43 to `"$VERSION"`

File: bin/publish (hunk 2 above)
Comment (doodlesbykumbi) [reply bin/publish:31]: `set -u` will complain if `TAG_NAME` is altogether not defined.

**Comment by izgeri** [comment]
@doodlesbykumbi do you want to add a version file to the repo? or will the version be indicated by the CHANGELOG?

we can add a version file later, too - perhaps in an effort to make sure it's easy for users to identify the sidecar injector version they're using.

**Comment by doodlesbykumbi** [comment]
@izgeri All set, version included. I'd initially thought to do it later but it really helps to be able to know the version I'm running.

**Comment by sgnn7** [comment]
@izgeri there's no set standard but generally we have kept just the main runner file in `cmd` in other projects and put version in `pkg/version.go`.

**Comment by izgeri** [comment]
@doodlesbykumbi wdyt about moving it to `pkg/inject/version.go`?

**Comment by doodlesbykumbi** [comment]
@izgeri  I suppose it makes the version available through pkg. Made the change

**Pull request merged**